wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java

```diff
@@ -1339,9 +1338,122 @@ public int pathUnlinkFile(int dirFd, @Buffer String rawPath) {
+                    int fd = memory.readInt(inPtr);
+                    if (fd < 0) {
+                        return wasiResult(WasiErrno.EBADF);
+                    }
+                    var descriptor = descriptors.get(fd);
+                    if (descriptor instanceof InStream && eventType == WasiEventType.FD_READ) {
+                        readEvents.add(Map.entry((InStream) descriptor, userData));
+                    } else if (descriptor instanceof OutStream
+                            && eventType == WasiEventType.FD_WRITE) {
+                        // assume output streams are always writable
+                        writeEvent(memory, outPtr, userData, eventType, WasiErrno.ESUCCESS);
+                        outPtr += 32;
+                        nevents++;
+                    } else {
+                        WasiErrno errno;
+                        if (descriptor == null) {
+                            errno = WasiErrno.EBADF;
+                        } else if (descriptor instanceof OpenFile) {
+                            // per specification: this event always triggers for regular files
+                            errno = WasiErrno.ESUCCESS;
+                        } else {
+                            errno = WasiErrno.ENOTSUP;
+                        }
+                        writeEvent(memory, outPtr, userData, eventType, errno);
+                        outPtr += 32;
+                        nevents++;
+                    }
+                    break;
+                default:
+                    return wasiResult(WasiErrno.EINVAL);
+            }
+            inPtr += 32;
+        }
+
+        long minTimeout = clockEvents.stream().mapToLong(Entry::getKey).min().orElse(0);
+        long start = System.nanoTime();
+        while (true) {
+            for (var entry : readEvents) {
+                InStream stream = entry.getKey();
+                long userData = entry.getValue();
+                try {
+                    int available = stream.available();
+                    if (available <= 0) {
+                        continue;
+                    }
+                    writeEvent(memory, outPtr, userData, WasiEventType.FD_READ, WasiErrno.ESUCCESS);
+                    memory.writeLong(outPtr + 16, available);
+                    outPtr += 32;
+                    nevents++;
+                } catch (IOException e) {
+                    writeEvent(memory, outPtr, userData, WasiEventType.FD_READ, WasiErrno.EIO);
+                    outPtr += 32;
+                    nevents++;
+                }
+            }
+            if (nevents > 0) {
+                break;
+            }
+
+            long elapsed = System.nanoTime() - start;
+            long duration = max(minTimeout, 0) - elapsed;
+            if (!readEvents.isEmpty()) {
+                duration = min(duration, MILLISECONDS.toNanos(100));
+            }
+            try {
+                NANOSECONDS.sleep(duration);
```

Comment:
Looking at the implementation of the method, I think we can use the underlying:

```java
Thread.sleep(0, duration);
```

Reply:
This doesn't work as nanoseconds must be less than one second. Also, it's less clear, as the reader needs to be familiar with the method to know that the second argument is nanoseconds.

Reply:
:+1: